care/facility/api/viewsets/patient_consultation.py

```diff
@@ -40,6 +40,14 @@ def list(self, request, *args, **kwargs):
         """
         return super().list(request, *args, **kwargs)
 
+    def perform_create(self, serializer):
+        try:
+            if not self.request.data["admitted"]:
+                self.request.data["bed_number"] = ""
+        except KeyError:
+            self.request.data["bed_number"] = ""
+        serializer.save()
```

Comment:
`perform_create` is called after validation using serializer. Also serializer will be handle a good amount of security things like SQL injection, etc.  Hence we should not be touching the`self.request.data` and only be using `serializer.validated_data`. 

Also conventionally such validation / data cleaning should be part of the `.validation()` of the serializer. Please move this function to the serializer.